### DISCUSSION TITLE
Number extension rewrite

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2900] [Suggestions]
1. Need to convert `registerFunction` / `registerOperator` to `e2function` syntax. Will make it easier to convert to future api later.
2. Adjust op costs
3. Add some constants? (`_DEG2RAD` / `_RAD2DEG`)
4. Deprecate redundant functions `inf()` / `pi()` / `e()` (They have constants)
5. Rewrite min/max functions to use native lua `math.min` / `math.max` (take variadic arguments and JIT)

I'll have a PR for this soon. Supercedes https://github.com/wiremod/wire/discussions/2899

**Comment by Denneisk** [comment]
Lump #2848 in with this?

**Reply by vurvdev** [reply]
Would severely complicate the PR unless we did some ugly hardcoding

**Comment by Denneisk** [comment]
Completed in #2903
